src/Symfony/Bundle/TwigBundle/Resources/config/twig.php

```diff
@@ -78,8 +80,17 @@
         ->set('twig.template_iterator', TemplateIterator::class)
             ->args([service('kernel'), abstract_arg('Twig paths'), param('twig.default_path'), abstract_arg('File name pattern')])
 
+        ->set('twig.template_cache.runtime_cache', FilesystemCache::class)
+            ->args([param('kernel.cache_dir').'/twig', abstract_arg('Twig filesystem cache options')])
+
+        ->set('twig.template_cache.readonly_cache', ReadOnlyFilesystemCache::class)
+            ->args([param('kernel.build_dir').'/twig'])
+
+        ->set('twig.template_cache.chain', ChainCache::class)
+            ->args([[service('twig.template_cache.runtime_cache'), service('twig.template_cache.readonly_cache')]])
```

Comment:
I'm likely missing something, but I failed to see the toggle point where the runtime filesystem strategy is skipped in favor of the readonly one. Can you elaborate about it?

Reply:
The runtime strategy (which uses the `cache_dir`) will start with an empty storage, because the CacheWarmer only warms the `build_dir`. So except if `autoReload` is enabled for twig, the `load` call on the runtime filesystem will not find the file, so the class will not exists and the read-only filesystem will be called.

At least, that's my understanding of the Cache mechanism in Twig.